### ISSUE TITLE
HaskellC2Hs fails with clang: error: no input files

### ISSUE THREAD

**Issue by aherrmann** [tweag/rules_haskell#1486]
**Describe the bug**
The `HaskellC2Hs` action of `c2hs_library` fails on macOS in bindist mode with the error
```
clang: error: no input files
c2hs: Error during preprocessing custom header file
```

Example occurrence: [CI log](https://github.com/tweag/rules_haskell/runs/1856347107#step:6:149), [BuildBuddy](https://app.buildbuddy.io/invocation/05d813bb-393f-4329-b21b-4e3cfc2cc888?target=//tests/repl-targets:hs-lib@repl)

**To Reproduce**
Run the following command on macOS in bindist mode
```
$ bazel build //tests/repl-targets:chs
```

**Expected behavior**
The build should succeed.

**Environment**
 - OS name + version: Mac OS X 10.15.7 19H114
 - Bazel version: 3.3.1
 - Version of the rules: 4d0d4e7ddbb8d56cf0138606c5a59b5705ad8c42 (based on master at 6e05343b196c396e55352b97c83a121c2d2f85ff)

**Additional context**

I can reproduce the issue on a macOS machine locally. Jumping into the sandbox dir (`bazel ... --sandbox_debug`) I can boil it down to this cpp invocation that fails in the same way
```
/usr/bin/cpp -E -x c -DC2HS_VERSION_MAJOR=0 -DC2HS_VERSION_MINOR=28 -DC2HS_VERSION_REV=6 bazel-out/darwin-fastbuild/bin/tests/repl-targets/chs-chs/Chs.chs.h
```
The file exists and only contains
```
#define C2HS_MIN_VERSION(mj,mn,rv) ((mj)<=C2HS_VERSION_MAJOR && (mn)<=C2HS_VERSION_MINOR && (rv)<=C2HS_VERSION_REV)
```

**Comment by Xophmeister** [comment]
Tests are explicitly skipped in 55f2c1c; that relegates it into "wontfix" territory. I'm not sure about that, but feel that it's nonetheless very low priority.

**Comment by avdv** [comment]
Calling the C pre-processor with `-E` and `-x c` seems weird to me. It only makes sense if you are calling the clang driver, but not if you already have the C pre-processor at hand.

Calling `/usr/bin/cpp -DC2HS_VERSION_MAJOR=0 -DC2HS_VERSION_MINOR=28 -DC2HS_VERSION_REV=6 Chs.chs.h` works just fine.

**Comment by aherrmann** [comment]
> Calling the C pre-processor with `-E` and `-x c` seems weird to me. It only makes sense if you are calling the clang driver, but not if you already have the C pre-processor at hand.

Good point, I think this may just be blindly copied from some place. Does changing this resolve the issue?
